src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -188,8 +188,25 @@ bool TryCreateParentDirectory(const string& prefix, const string& filename) {
 // Get the absolute path of this protoc binary.
 bool GetProtocAbsolutePath(string* path) {
 #ifdef _WIN32
-  char buffer[MAX_PATH];
-  int len = GetModuleFileName(NULL, buffer, MAX_PATH);
+  wchar_t wide_buffer[MAX_PATH];
+  // Note: the returned length does not include terminating null character.
+  int len = GetModuleFileNameW(NULL, wide_buffer, MAX_PATH);
```

Comment:
@TeBoring can you give me a pointer to the discussion where people want to use GetModuleFileNameA here?

@xfxyjwf, I don't say we should not use char array.  The best way here is to have a wrapper function for GetModuleFileName so that we can retrieve the path as UTF-8 in a char array.  My point is we should use wchar_t array as an intermediate buffer because wchar_t is the standard data type in Win32 APIs.